src/datadog/trace_segment.cpp

```diff
@@ -337,9 +362,9 @@ bool TraceSegment::inject(DictWriter& writer, const SpanData& span,
         if (origin_) {
           writer.set("x-datadog-origin", *origin_);
         }
-        if (opts.delegate_sampling_decision) {
-          expecting_response = true;
-          expecting_delegated_sampling_decision_ = true;
+        if (options.delegate_sampling_decision) {
+          delegated_trace_sampling_decision =
+              sampling_delegation_.sent_request_header = true;
```

Comment:
Oops. It would be better if I didn't use assignment chaining:
```c++
    delegated_trace_sampling_decision = true;
    sampling_delegation_.sent_request_header = true;
```